### PR TITLE
Enable IDE0029 (Use coalesce expression)

### PR DESCRIPTION
**Pull request by stephentoub**

### CODE REVIEW

File: src/libraries/System.Data.Common/src/System/Data/Common/DBCommandBuilder.cs [hunk 1]
```diff
@@ -477,8 +477,7 @@ public virtual string QuoteSuffix
         {
             get
             {
-                string? quoteSuffix = _quoteSuffix;
-                return ((null != quoteSuffix) ? quoteSuffix : string.Empty);
+                return (_quoteSuffix ?? string.Empty);
```
Comment (danmoseley) [inline src/libraries/System.Data.Common/src/System/Data/Common/DBCommandBuilder.cs:480]: any particular reason for parens?

Comment (stephentoub) [reply src/libraries/System.Data.Common/src/System/Data/Common/DBCommandBuilder.cs:480]: Nope, they just aren't removed by the auto-fixer for the rule.  I removed them in cases like this as I was reviewing the changes if I noticed them.  There's a separate analyzer for removing unnecessary parens, which we're not ready for yet :)

File: src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlNodeReader.cs [hunk 2]
```diff
@@ -254,8 +256,10 @@ public string Value
                     retValue = strb.ToString();
                 }
                 else
+                {
                     retValue = _curNode.Value;
-                return (retValue == null) ? string.Empty : retValue;
+                }
+                return retValue ?? string.Empty;
```
Comment (danmoseley) [inline src/libraries/System.Private.Xml/src/System/Xml/Dom/XmlNodeReader.cs:262]: nit . new line?

File: src/libraries/System.Speech/src/Internal/Synthesis/AudioFormatConverter.cs [hunk 3]
```diff
@@ -252,7 +252,7 @@ private static byte[] CalcLinear2ULawTable()
         internal static byte[] ConvertLinear2ALaw(short[] data, int size)
         {
             byte[] newData = new byte[size];
-            s_aLawCompTableCached = s_aLawCompTableCached == null ? CalcLinear2ALawTable() : s_aLawCompTableCached;
+            s_aLawCompTableCached = s_aLawCompTableCached ?? CalcLinear2ALawTable();
```
Comment (danmoseley) [inline src/libraries/System.Speech/src/Internal/Synthesis/AudioFormatConverter.cs:255]:
```suggestion
            s_aLawCompTableCached ??= CalcLinear2ALawTable();
```

**Review by danmoseley — APPROVED** [review]

**Pull request merged**